### PR TITLE
fqdn: Use read-write mutex inside NameManager

### PR DESCRIPTION
**Pull request by christarazi**
Reviewing a trace pprof which was taken in an environment with a large
amount of FQDN policies, there was a non-trivial amount of goroutines
that were stuck on the NameManager mutex. Many of the calls were stuck
on the (*NameManager).GetModel() call which is a read-only operation.

We can fix this by moving to a read-write mutex and making GetModel()
acquire the read-lock.

Signed-off-by: Chris Tarazi <chris@isovalent.com>

### CODE REVIEW

**Review by nebril — APPROVED** [review]

**Comment by christarazi** [comment]
VM provisioning failures, retriggering Jenkins

File: pkg/fqdn/name_manager.go [hunk 1]
```diff
@@ -66,13 +66,13 @@ func (n *NameManager) GetModel() *models.NameManager {
 // Lock must be held during any calls to RegisterForIdentityUpdatesLocked or
 // UnregisterForIdentityUpdatesLocked.
 func (n *NameManager) Lock() {
```
Comment (ldelossa) [inline pkg/fqdn/name_manager.go:68]: @christarazi isn't this redundant? You embed the `lock.RWMutex` which promotes its methods up to the `NameManager` structure, no ?

**Review by ldelossa — CHANGES_REQUESTED** [review]

File: pkg/fqdn/name_manager.go (hunk 1 above)
Comment (christarazi) [reply pkg/fqdn/name_manager.go:68]: It is, but this code existed before this change and I didn't want to refactor everything else around the mutex when I just wanted to get to the optimization of using a rwmutex. I think the code could use a little clean up, but I don't think this PR should do it now.

Comment (ldelossa) [reply pkg/fqdn/name_manager.go:68]: Got it, didn't realize that

**Review by ldelossa — APPROVED** [review]

**Pull request merged**